app/uk/gov/hmrc/exports/services/reversemapping/declaration/items/SingleItemParser.scala

```diff
@@ -80,33 +88,68 @@ class SingleItemParser @Inject()(procedureCodesParser: ProcedureCodesParser) ext
   private def parseDangerousGoodsCode(itemXml: NodeSeq): Option[UNDangerousGoodsCode] =
     (itemXml \ Commodity \ DangerousGoods \ UNDGID).toStringOption.map(code => UNDangerousGoodsCode(Some(code)))
 
-  private def parseCusCode(itemXml: NodeSeq): Option[CUSCode] = {
-    val cusCodeTypeCode = "CV"
-
+  private def parseCusCode(itemXml: NodeSeq): Option[CUSCode] =
     filterClassificationIdsFor(itemXml, cusCodeTypeCode).headOption
       .map(cusCode => CUSCode(Some(cusCode)))
-  }
-
-  private def parseTaricCodes(itemXml: NodeSeq): Option[List[TaricCode]] = {
-    val taricCodeTypeCode = "TRA"
 
+  private def parseTaricCodes(itemXml: NodeSeq): Option[List[TaricCode]] =
     Some(
       filterClassificationIdsFor(itemXml, taricCodeTypeCode).toList
         .map(TaricCode(_))
     )
-  }
-
-  private def parseNactCodes(itemXml: NodeSeq): Option[List[NactCode]] = {
-    val nactCodeTypeCode = "GN"
 
+  private def parseNactCodes(itemXml: NodeSeq): Option[List[NactCode]] =
     Some(
       filterClassificationIdsFor(itemXml, nactCodeTypeCode).toList
         .map(NactCode(_))
     )
+
+  private def parseCommodityMeasure(inputXml: NodeSeq): Option[CommodityMeasure] = {
+    val commodityMeasure =
+      List((inputXml \ TariffQuantity).toStringOption, (inputXml \ NetNetWeightMeasure).toStringOption, (inputXml \ GrossMassMeasure).toStringOption)
```

Comment:
What would happen if the second element of the List is None, wouldn't element 1 then be what you thought was int element 2?